Update README (#50)
golemfactory/task-api: README.md

- `NextSubtask`
  - Takes two arguments: `task_id` and `opaque_node_id`.
  - `opaque_node_id` is an identifier of the node which is going to compute the requested subtask. 'Opaque' means that the identifier doesn't allow to obtain any further information about the node (e.g. public key, IP address).
  - Can assume `CreateTask` was called earlier with the same `task_id`.
  - Can return an empty message meaning that the app refuses to assign a subtask to the provider node (for whatever reason).
  - Returns `subtask_id` which has to be a string without whitespaces and slashes (`/`) but the same string cannot be returned more than once.
  - Also returns `subtask_params_json` which is the JSON string containing subtask specific parameters.
  - Also returns `resources` which is a list of names of files required for computing the subtask. Files with these names are required to be present in `{task_id}/{constants.SUBTASK_INPUTS_DIR}` directory.
- `HasPendingSubtasks`
  - Takes one argument `task_id`.
  - Returns a boolean indicating whether there are any more pending subtasks waiting for computation at given moment.

>   - In case when it returns `true`, the next `NextSubtask` call should return successfully (it can return an empty message altough).

Reviewer: nit: would move although to the start of the bracketed sentence

>   - In case when it returns `true`, the next `NextSubtask` call should return successfully (it can return an empty message altough).

Author: ```suggestion
  - In case when it returns `true`, the next `NextSubtask` call should return successfully (although it can still return an empty message).
```

- `Verify`
  - Takes two arguments: `task_id` and `subtask_id` which specify which subtask results should be verified.
  - Will be called with only valid `task_id` and `subtask_id` values.
  - Returns `result` which is one of the defined verification result statuses:
    - `SUCCESS` - the subtask was computed correctly,
    - `FAILURE` - the subtask was computed incorrectly,
    - `INCONCLUSIVE` - cannot determine whether the subtask was computed correctly,
    - `AWAITING_DATA` - cannot perform verificaton until results of other subtasks are available.
  - Also returns `reason` which is a string providing more detail about the result.

>   - For successfully verified subtasks it most likely should also perform merging the partial results into the final one.

Reviewer: ```suggestion
  - For successfully verified subtasks it can also perform merging of the partial results into the final one.
```

### RPC commands
- `Compute`
  - Gets a single working directory `task_work_dir` to operate on.
  - Different subtasks of the same task will have the same `task_work_dir`.
  - Takes `task_id`, `subtask_id`, `subtask_params_json` as arguments.
  - Can assume the `{task_work_dir}/{constants.SUBTASK_INPUTS_DIR}` directory exists.
  - Can assume that under `{task_work_dir}/{constants.SUBTASK_INPUTS_DIR}` are the resources specified in the corresponding `NextSubtask` call.
  - Returns a filepath (relative to the `task_work_dir`) of the result file which will be sent back to the requestor with unchanged name.
- `Benchmark`
  - Takes no arguments.
  - Returns a score which indicates how efficient the machine is for this type of tasks.
  - Shouldn't take much time (preferably less than a minute for medium range machines).
- `Shutdown`
  - Takes no arguments.
  - Should gracefully terminate the service.

>   - Will be called in case the provider interrupts task computation or benchmark.

Reviewer: ```suggestion
  - Can be called in case the provider wants to interrupt task computation or benchmark.
```
